src/main/webapp/app/entities/study/study-detail.component.ts

```diff
@@ -25,11 +28,26 @@ export class StudyDetailComponent implements OnInit, OnDestroy {
     ) {
     }
 
+    onButtonClicked(currentStatus:boolean){
+        if (this.presetSend === currentStatus){
+            this.presetSend = !this.presetSend;
+        }
+    }
+
     ngOnInit() {
         this.subscription = this.route.params.subscribe((params) => {
             this.load(params['id']);
         });
         this.registerChangeInStudies();
+        this.currentTimeStamp = StudyService.getCurrentDateTime();
+        this.currentTimeZone = this.getTimeZone();
+    }
+
+    // credit for function: https://stackoverflow.com/questions/1091372/getting-the-clients-timezone-in-javascript
+    // User: Mr_Green
+    getTimeZone(): string {
+        var offset = new Date().getTimezoneOffset(), o = Math.abs(offset);
```

Comment:
Try using const and let in TS. It is not a good idea to mix JS into TS.
'var offset'

Reply:
thanks, fixed.